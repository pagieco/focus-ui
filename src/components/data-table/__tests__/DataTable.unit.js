import { mount, shallowMount } from '@vue/test-utils';
import DataTable from '../components/DataTable.vue';

const shallowWrap = (propsData = {}) => shallowMount(DataTable, { propsData });
const wrap = (propsData = {}) => mount(DataTable, { propsData });

const headings = ['Product', 'Price', 'Order Number', 'Quantity', 'Subtotal'];

const rows = [
  [{ content: 'Emerald Silk Gown', url: '#' }, '$875.00', 124689, 140, '$122,500.00'],
  ['Navy Merino Wool Blazer with khaki chinos and yellow belt', '$445.00', 124518, 32, '$14,240.00'],
  ['Mauve Cashmere Scarf', '$230.00', 124533, 83, '$19,090.00'],
];

const columnContentTypes = [
  'text',
  'numeric',
  'numeric',
  'numeric',
  'numeric',
];

describe('DataTable', () => {
  it('is a valid vue component', () => {
    expect(DataTable).toBeAComponent();
  });

  it('matches the snapshot', () => {
    expect(shallowWrap({
      columnContentTypes,
      headings,
      rows,
    }).element).toMatchSnapshot();
  });

  it('uses the columnContentTypes prop', () => {
    const wrapper = wrap({ columnContentTypes, headings, rows });

    expect(wrapper.findAll('thead .data-table__cell--text').length).toBe(1);
    expect(wrapper.findAll('thead .data-table__cell--numeric').length).toBe(4);
  });

  it('validates the columnContentTypes prop', () => {
    const prop = DataTable.props.columnContentTypes;

    expect(prop.validator).toBeInstanceOf(Function);
    expect(prop.validator(['text', 'numeric'])).toBeTruthy();
    expect(prop.validator(['text', 'image'])).toBeFalsy();
  });

  it('uses the defaultSortDirection prop', () => {

  });

  it('validates the defaultSortDirection', () => {

  });

  it('ues the footerContent prop', () => {
    const footerContent = `Showing ${rows.length} of ${rows.length} records`;
    const wrapper = wrap({
      columnContentTypes,
      headings,
      rows,
      footerContent,
    });

    expect(wrapper.find('.data-table__footer').exists()).toBeTruthy();
  });

  it('uses the headings prop', () => {
    const wrapper = wrap({ columnContentTypes, headings, rows });

    wrapper.findAll('thead .data-table__cell').wrappers.forEach((w) => {
      expect(headings.includes(w.text())).toBeTruthy();
    });
  });

  it('uses the initialSortColumnIndex prop', () => {

  });

  it('uses the rows prop', () => {
    const wrapper = wrap({ columnContentTypes, headings, rows });

    expect(wrapper.findAll('tbody tr').length).toBe(rows.length);
  });

  it('uses the sortable prop', () => {

  });

  it('uses the totals prop', () => {

  });
});
