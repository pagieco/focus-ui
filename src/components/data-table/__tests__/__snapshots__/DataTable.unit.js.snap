// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable matches the snapshot 1`] = `
<div
  class="data-table"
>
  <table
    class="data-table__table"
  >
    <thead>
      <tr>
        <cell-stub
          content="Product"
          contenttype="text"
          firstcolumn="true"
          header="true"
          scope="col"
        />
        <cell-stub
          content="Price"
          contenttype="numeric"
          header="true"
          scope="col"
        />
        <cell-stub
          content="Order Number"
          contenttype="numeric"
          header="true"
          scope="col"
        />
        <cell-stub
          content="Quantity"
          contenttype="numeric"
          header="true"
          scope="col"
        />
        <cell-stub
          content="Subtotal"
          contenttype="numeric"
          header="true"
          scope="col"
        />
      </tr>
       
      <tr />
    </thead>
     
    <tbody>
      <tr
        class="data-table__row"
      >
        <cell-stub
          content="[object Object]"
          contenttype="text"
          firstcolumn="true"
        />
        <cell-stub
          content="$875.00"
          contenttype="numeric"
        />
        <cell-stub
          content="124689"
          contenttype="numeric"
        />
        <cell-stub
          content="140"
          contenttype="numeric"
        />
        <cell-stub
          content="$122,500.00"
          contenttype="numeric"
        />
      </tr>
      <tr
        class="data-table__row"
      >
        <cell-stub
          content="Navy Merino Wool Blazer with khaki chinos and yellow belt"
          contenttype="text"
          firstcolumn="true"
        />
        <cell-stub
          content="$445.00"
          contenttype="numeric"
        />
        <cell-stub
          content="124518"
          contenttype="numeric"
        />
        <cell-stub
          content="32"
          contenttype="numeric"
        />
        <cell-stub
          content="$14,240.00"
          contenttype="numeric"
        />
      </tr>
      <tr
        class="data-table__row"
      >
        <cell-stub
          content="Mauve Cashmere Scarf"
          contenttype="text"
          firstcolumn="true"
        />
        <cell-stub
          content="$230.00"
          contenttype="numeric"
        />
        <cell-stub
          content="124533"
          contenttype="numeric"
        />
        <cell-stub
          content="83"
          contenttype="numeric"
        />
        <cell-stub
          content="$19,090.00"
          contenttype="numeric"
        />
      </tr>
    </tbody>
  </table>
   
  <!---->
</div>
`;
